src/Dictionaries/DictionarySourceFactory.cpp

```diff
@@ -84,8 +84,9 @@ DictionarySourcePtr DictionarySourceFactory::create(
 {
     Poco::Util::AbstractConfiguration::Keys keys;
     config.keys(config_prefix, keys);
-    if (keys.size() != 1)
-        throw Exception{name + ": element dictionary.source should have exactly one child element",
+
+    if (keys.empty() || keys.size() > 2)
+        throw Exception{name + ": element dictionary.source should have one or two child elements",
                         ErrorCodes::EXCESSIVE_ELEMENT_IN_CONFIG};
 
     const auto & source_type = keys.front();
```

Comment:
@tavplubix unfortunately, I don't understand what you're meaning

Reply:
`<source>` element may have two child elements: `<source_type>` (e.g. `<file>` or `<http>`) and `<settings>`. If it have only `<source_type>`, then `keys.front()` will obliviously return name of dictionary source, because there is only one key. But if there is also `<settings>` then `keys.front()` may return `settings` instead of the dictionary source name, so you will get exception like `unknown dictionary source type: settings`